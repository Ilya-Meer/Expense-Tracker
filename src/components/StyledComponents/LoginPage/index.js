import styled from 'styled-components';


const LoginPageWrapper = styled.div`
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(rgba(255,255,255,.2), rgba(255,255,255,.2)), url('/images/loginbg.jpg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
`

const Dialog = styled.div`
  height: 400px;
  width: 85%;
  padding: 3rem 1.5rem;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  color: #333;
  background: rgba(250, 250, 250, 0.7);
  box-shadow: rgba(0, 0, 0, 0.2) 0px 10px 30px 0px;
  @media all and (min-width: 620px) {
    width: 350px;
  }

  h1 {
    text-align: center;
  }

  p {
    margin: 5px 0; 
  }

  button {
    margin: 1rem 0 0 0;
    border: 1px solid rgba(255,255,255,0.2); 
  }
`

export { LoginPageWrapper, Dialog };