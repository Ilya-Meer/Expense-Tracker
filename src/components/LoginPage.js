import React from 'react';
import { connect } from 'react-redux';
import { FaGoogle } from 'react-icons/fa';
import { LoginPageWrapper, Dialog } from './StyledComponents/LoginPage';
import { Button } from './StyledComponents/Shared';
import { startLogin } from '../actions/authActions';

const LoginPage = (props) => (
  <LoginPageWrapper>
    <Dialog>  
      <h1>Expense Tracker</h1>
      <p>Follow your money.</p>
      <p>Control your expenses.</p>
      <Button 
        onClick={props.startLogin}>
        <FaGoogle />Log In with Google
      </Button>
    </Dialog>
  </LoginPageWrapper>
);

export { LoginPage };



const mapDispatchToProps = (dispatch) => {
  return {
    startLogin: () => {
      dispatch(startLogin())
    }
  }
}

export default connect(undefined, mapDispatchToProps)(LoginPage)