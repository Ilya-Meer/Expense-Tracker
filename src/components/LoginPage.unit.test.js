import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from './LoginPage';


describe('LoginPage', () => {
  it('should render the Login Page', () => {
    const wrapper = shallow(<LoginPage startLogin={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  })
})

describe('Login Button', () => {
  it('should log user in when log in button is clicked', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
    wrapper.find('button').at(0).simulate('click');
    expect(startLogin).toHaveBeenCalled();
  })
})
