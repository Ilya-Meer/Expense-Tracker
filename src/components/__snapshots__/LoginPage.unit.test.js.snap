// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render the Login Page 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Log In
  </button>
  <button>
    Sign Up
  </button>
</div>
`;
